python-package/lightgbm/basic.py

```diff
@@ -156,6 +156,28 @@ def _safe_call(ret: int) -> None:
         raise LightGBMError(_LIB.LGBM_GetLastError().decode('utf-8'))
 
 
+def _get_string_from_c_api(func: Callable, booster_handle: Optional[ctypes.c_void_p] = None) -> str:
```

Comment:
Could you please make a separate PR with this code deduplication change only?
We have other places with string buffer creation, e.g. in
https://github.com/microsoft/LightGBM/blob/e063dad261d06049a461c3e2a364350e2b20b9f7/python-package/lightgbm/basic.py#L2315
https://github.com/microsoft/LightGBM/blob/e063dad261d06049a461c3e2a364350e2b20b9f7/python-package/lightgbm/basic.py#L3510
https://github.com/microsoft/LightGBM/blob/e063dad261d06049a461c3e2a364350e2b20b9f7/python-package/lightgbm/basic.py#L3568
https://github.com/microsoft/LightGBM/blob/e063dad261d06049a461c3e2a364350e2b20b9f7/python-package/lightgbm/basic.py#L3865
https://github.com/microsoft/LightGBM/blob/e063dad261d06049a461c3e2a364350e2b20b9f7/python-package/lightgbm/basic.py#L4085

Reply:
Removed in 0e6591b. d4e781b also adds a lot of duplicated for string buffers in `lightgbm_R.cpp`, could address that in a follow up PR as well.